log-collector/src/test/scala/io/phdata/pulse/logcollector/LogCollectorRoutesTest.scala

```diff
@@ -154,4 +155,21 @@ class LogCollectorRoutesTest
       responseAs[String] == "Log Collector"
     }
   }
+
+  test("return 500 on bad request to kudu") {
+    val request = TimeseriesRequest("metrics", List(TimeseriesEvent(1, "key", "tag", 1.4)))
+    val entity  = HttpEntity(ContentTypes.`application/json`, request.toJson.toString())
+
+    when(kuduService.save(Matchers.any(), Matchers.any()))
+      .thenThrow(new RuntimeException("Save failed"))
+
+    Post(uri = "/v1/metrics")
+      .withEntity(entity) ~> routes ~> check {
+      val caught = intercept[RuntimeException] {
+        kuduService.save(Matchers.any(), Matchers.any())
+      }
```

Comment:
You don't want to call `kuduService.save` yourself, when you call the route with a request entity the route calls `save` already.

You can just remove the intercept because I don't think that works with the Akka routes testing, just check for the 500 error.

```
val caught = intercept[RuntimeException] {
    Post(uri = "/v1/metrics")
      .withEntity(entity) ~> routes ~> check {
      
        kuduService.save(Matchers.any(), Matchers.any())
      }
      assert(caught.getMessage === "Save failed" && status === StatusCodes.InternalServerError)
    }

```